src/native/libs/System.Security.Cryptography.Native.Apple/CMakeLists.txt

```diff
@@ -42,16 +42,38 @@ endif()
 # As of CMake 3.20.2 support for Swift only works with the Ninja and XCode
 # generators so we cannot rely on it. Even with the Ninja generator it doesn't
 # work in combination with other languages within the same library.
-if (NOT SWIFT_COMPILER_TARGET AND CLR_CMAKE_TARGET_OSX)
-    set(SWIFT_PLATFORM "macosx")
-    set(SWIFT_PLATFORM_SUFFIX "")
+if (NOT SWIFT_COMPILER_TARGET)
     set(SWIFT_DEPLOYMENT_TARGET ${CMAKE_OSX_DEPLOYMENT_TARGET})
+    set(SWIFT_PLATFORM_SUFFIX "")
+
+    if (CMAKE_BUILD_TYPE STREQUAL "Debug")
+        set(SWIFT_OPTIMIZATION_FLAG "-Onone")
+    elseif (CMAKE_BUILD_TYPE STREQUAL "Release")
+        set(SWIFT_OPTIMIZATION_FLAG "-O")
+    endif()
```

Comment:
I noticed that we didn't compile the swift code with optimizations so fixed this as well